main/options/options.cc

```diff
@@ -800,6 +802,25 @@ void readOptions(Options &opts,
         opts.metricsPrefix = raw["metrics-prefix"].as<string>();
         opts.debugLogFile = raw["debug-log-file"].as<string>();
         opts.webTraceFile = raw["web-trace-file"].as<string>();
+        {
+            // parse extra sfx/datadog tags
+            auto stringToParse = raw["metrics-extra-tags"].as<string>();
+            if (stringToParse != "") {
+                size_t pos = 0;
+                vector<string> pairs;
+                while ((pos = stringToParse.find(',')) != std::string::npos) {
+                    pairs.emplace_back(stringToParse.substr(0, pos));
+                    stringToParse.erase(0, pos + 1);
+                }
+                pairs.emplace_back(stringToParse); // last one
+                for (auto &pair : pairs) {
+                    pos = pair.find('=');
+                    auto key = pair.substr(0, pos);
+                    pair.erase(0, pos + 1); // keep only value
+                    opts.metricsExtraTags[key] = pair;
+                }
+            }
+        }
```

Comment:
This is actually the textbook example from the `absl::StrSplit` method:

```cpp
// The input string "a=b=c,d=e,f=,g" becomes
// { "a" => "b=c", "d" => "e", "f" => "", "g" => "" }
std::map<std::string, std::string> m;
for (absl::string_view sp : absl::StrSplit("a=b=c,d=e,f=,g", ',')) {
  m.insert(absl::StrSplit(sp, absl::MaxSplits('=', 1)));
}
```

in case you wanted to shorten this and write less fiddly index code.

Reply:
Done